src/sandstorm/run-bundle.c++

```diff
@@ -921,8 +921,15 @@ private:
     return kj::heapString(exeName.slice(0, KJ_ASSERT_NONNULL(exeName.findLast('/'))));
   }
 
-  void changeToInstallDir() {
+  void changeToInstallDirAndCheckAccess() {
     KJ_SYSCALL(chdir(getInstallDir().cStr()));
+    if (access("../var/sandstorm", W_OK) == -1) {
+      if (errno == EACCES) {
+        KJ_FAIL_REQUIRE("Sandstorm must be run as root or a member of SERVER_USER's group.");
```

Comment:
This error is not quite correct:
- If Sandstorm was installed as root, then it must be started as root, regardless of SERVER_USER.
- If Sandstorm was installed as non-root, then it must always be run as the same user who installed it, and I think SERVER_USER is not actually specified in the config.